src/Mailgun/Mailgun.php

```diff
@@ -47,6 +60,51 @@ public function __construct(
     ) {
         $this->apiKey = $apiKey;
         $this->restClient = new RestClient($apiKey, $apiEndpoint, $httpClient);
+
+        $this->initializeRegionClients($httpClient);
+    }
+
+    /**
+     * This function creates clients for each individual region. Messages are
+     * only accessible from the region that received them, so having a mapping
+     * of region -> client seems reasonable.
+     *
+     * @param HttpClient|null $httpClient
+     *
+     * @return bool
+     */
+    private function initializeRegionClients(HttpClient $httpClient = null)
+    {
+        foreach ($this->regions as $region => $apiUrl) {
+            $this->regionClients[$region] = new RestClient(
+                $this->apiKey,
+                $apiUrl,
+                $httpClient
+            );
+        }
+
+        return true;
```

Comment:
❓ Question: Why to add here `return true;`? If we are not going to return any other value does not make sense - from my POV - to return always `true` so we can just remove this. Anyway, if we throw an Exception (for example) we are not going to reach that `return` so IMHO does not make sense or is not adding a useful case.